### PR TITLE
Added an extended PCA example with the iris dataset and plotting

### PR DESCRIPTION
**Pull request by ianshmean**
I could've done with a more fleshed-out PCA example, so here's a suggestion

### CODE REVIEW

File: docs/source/pca.rst [hunk 1]
```diff
@@ -142,6 +142,46 @@ One can use the ``fit`` method to perform PCA over a given dataset.
...
+
+    # split other half to testing set
+    Xte = convert(Array,DataArray(iris[2:2:end,1:4]))'
+    Xte_labels = convert(Array,DataArray(iris[2:2:end,5]))
+
+    # suppose Xtr and Xte are training and testing data matrix,
+    # with each observation in a column
+
+    # train a PCA model
+    M = fit(PCA, Xtr; maxoutdim=100)
```
Comment (wildart) [inline docs/source/pca.rst]: You can just call `M = fit(PCA, Xtr, maxoutdim=3)` to get first 3 principal components.

File: docs/source/pca.rst [hunk 2]
```diff
@@ -142,6 +142,46 @@ One can use the ``fit`` method to perform PCA over a given dataset.
...
+
+    # apply PCA model to testing set
+    Yte = transform(M, Xte)
+
+    # reconstruct testing observations (approximately)
+    Xr = reconstruct(M, Yte)
+
+    # group results by testing set labels for color coding, and select first 3 dimensions
+    setosa = Yte[1:3,Xte_labels.=="setosa"]
```
Comment (wildart) [inline docs/source/pca.rst]: and then `Yte[:,Xte_labels.=="setosa"]` and so on.

File: docs/source/pca.rst (hunk 1 above)
Comment (ianshmean) [reply docs/source/pca.rst]: OK, just updated

**Comment by lwabeke** [comment]
This example doesn't seem to work for me on Julia 0.6.2:
did a `Pkg.checkout("MultivariateStats")`
I get an error:
UndefVarError: DataArray not defined

Adding
`using DataArrays`
I get the error:
MethodError: Cannot `convert` an object of type DataFrames.DataFrame to an object of type DataArrays.DataArray
This may have arisen from a call to the constructor DataArrays.DataArray(...),
since type constructors fall back to convert methods.

Suggestions on what I'm missing?

**Comment by wildart** [comment]
Try not to use DataArray
```julia
# split half to training set
Xtr = convert(Array, iris[1:2:end,1:4])'
Xtr_labels = convert(Array, iris[1:2:end,5])

# split other half to testing set
Xte = convert(Array, iris[2:2:end,1:4])'
Xte_labels = convert(Array, iris[2:2:end,5])
```

**Pull request merged**